test/metabase/driver/mysql_test.clj

```diff
@@ -12,6 +15,30 @@
 
 (expect-with-engine :mysql
   [[1 nil]]
+  ;; TODO - use the `rows` function from `metabse.query-processor-test`. Preferrably after it's moved to some sort of shared test util namespace
   (-> (data/dataset metabase.driver.mysql-test/all-zero-dates
         (data/run-query exciting-moments-in-history))
       :data :rows))
+
+
+;;; connection details
+
+;; make sure connection details w/o extra params work normally
+(expect
+  {:classname   "com.mysql.jdbc.Driver"
+   :subprotocol "mysql"
+   :subname     "//localhost:3306/cool?zeroDateTimeBehavior=convertToNull&useUnicode=true&characterEncoding=UTF8&characterSetResults=UTF8&useSSL=false"
+   :delimiters  "`"}
+  (generic-sql/connection-details->spec (MySQLDriver.) {:host   "localhost"
+                                                        :port   "3306"
+                                                        :dbname "cool"}))
+
+;; make sure if extra parameters are included that the correct combos of ?/& are used
+(expect
+  {:classname   "com.mysql.jdbc.Driver"
+   :subprotocol "mysql"
+   :subname     "//localhost:3306/cool?tinyInt1isBit=false&zeroDateTimeBehavior=convertToNull&useUnicode=true&characterEncoding=UTF8&characterSetResults=UTF8&useSSL=false"
+   :delimiters  "`"}
+  (generic-sql/connection-details->spec (MySQLDriver.) {:host   "localhost"
+                                                        :port   "3306"
+                                                        :dbname "cool?tinyInt1isBit=false"}))
```

Comment:
One single guy who had a really bad schema is going to have to do it here to work around the default JDBC driver behavior.